File: src/main/java/work/lclpnet/ap2/game/chicken_shooter/ChickenShooterMiniGame.java
package work.lclpnet.ap2.game.chicken_shooter;

import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.util.Identifier;
import work.lclpnet.ap2.api.game.*;
import work.lclpnet.ap2.base.ApConstants;
import work.lclpnet.ap2.base.ArcadeParty;

public class ChickenShooterMiniGame implements MiniGame {

    @Override
    public Identifier getId() {
        return ArcadeParty.identifier("chicken_shooter");
    }

    @Override
    public GameType getType() {
        return GameType.FFA;
    }

    @Override
    public String getAuthor() {
        return ApConstants.PERSON_BOPS;
    }

    @Override
    public ItemStack getIcon() {
        return new ItemStack(Items.FEATHER);
    }

    @Override
    public boolean canBeFinale(GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(GameStartContext context) {
        return true;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new ChickenShooterInstance(gameHandle);
    }
}

File: src/minigames/chicken_shooter/src/main/kotlin/work/lclpnet/ap2/game/chicken_shooter/ChickenShooterMiniGame.kt
package work.lclpnet.ap2.game.chicken_shooter

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.base.GameStartContext
import work.lclpnet.ap2.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.MiniGameFactory
import work.lclpnet.ap2.game.util.MapLevelGameFactory

class ChickenShooterMiniGame : MiniGame {
    override val id = ApConstants.identifier("chicken_shooter")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_BOPS
    override fun getIcon(manager: RegistryAccess): ItemStack = ItemStack(Items.FEATHER)
    override fun canBeFinale(context: GameStartContext): Boolean = true
    override fun canBePlayed(context: GameStartContext): Boolean = true
    override fun createFactory(): MiniGameFactory = MapLevelGameFactory(::ChickenShooterInstance)
}
